doc/src/manual/arrays.md

```diff
@@ -764,9 +786,12 @@ julia> speye(3,5)
   [3, 3]  =  1.0
 ```
 
-The [`sparse()`](@ref) function is often a handy way to construct sparse matrices. It takes as
-its input a vector `I` of row indices, a vector `J` of column indices, and a vector `V` of stored
-values. `sparse(I,J,V)` constructs a sparse matrix such that `S[I[k], J[k]] = V[k]`.
+The [`sparse()`](@ref) function is often a handy way to construct sparse array. For
+matrices it takes as its input a vector `I` of row indices, a vector `J` of column
+indices, and a vector `V` of stored values. `sparse(I,J,V)` constructs a sparse matrix
+such that `S[I[k], J[k]] = V[k]`. The equivalent sparse vector constructor is
+[`sparsevec`](@ref), which takes one vector `I` with indices and one vector `V` with the
```

Comment:
This made me think we should deprecate `sparsevec` in favor of `sparse`? There is already inconsistent behavior, sometimes `sparse` returns a `SparseVector`, e.g. in `sparse(rand(3))`. WIP at 75bc508aa00723c998673c47c8f0ca22c8a272f1

The only problem (which is all thats causing test fails on my WIP commit) is the following method `sparse(a::Vector, b::Vector, c::Int)` which could either be a `SparseVector` where the `c` is the length, or a `SparseMatrixCSC` which should be filled with the value `c`. Is there an elegant way to possibly get around this? :)